doc/snippets/Microsoft.Data.SqlClient/SqlConfigurableRetryFactory.xml

```diff
@@ -0,0 +1,119 @@
+|49919|Cannot process create or update request. Too many create or update operations in progress for subscription "%ld".|
+|49920|Cannot process request. Too many operations in progress for subscription "%ld".|
+|4060|Cannot open database "%.*ls" requested by the login. The login failed.|
+|4221|Login to read-secondary failed due to long wait on 'HADR_DATABASE_WAIT_FOR_TRANSITION_TO_VERSIONING'. The replica is not available for login because row versions are missing for transactions that were in-flight when the replica was recycled. The issue can be resolved by rolling back or committing the active transactions on the primary replica. Occurrences of this condition can be minimized by avoiding long write transactions on the primary.|
+|40143|The service has encountered an error processing your request. Please try again.|
+|40613|Database '%.*ls' on server '%.*ls' is not currently available. Please retry the connection later. If the problem persists, contact customer support, and provide them the session tracing ID of '%.*ls'.|
+|40501|The service is currently busy. Retry the request after 10 seconds. Incident ID: %ls. Code: %d.|
+|40540|The service has encountered an error processing your request. Please try again.|
+|40197|The service has encountered an error processing your request. Please try again. Error code %d.|
+|10929|Resource ID: %d. The %s minimum guarantee is %d, maximum limit is %d and the current usage for the database is %d. However, the server is currently too busy to support requests greater than %d for this database. For more information, see http://go.microsoft.com/fwlink/?LinkId=267637. Otherwise, please try again later.|
+|10928|Resource ID: %d. The %s limit for the database is %d and has been reached. For more information, see http://go.microsoft.com/fwlink/?LinkId=267637.|
+|10060|An error has occurred while establishing a connection to the server. When connecting to SQL Server, this failure may be caused by the fact that under the default settings SQL Server does not allow remote connections. (provider: TCP Provider, error: 0 - A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.) (Microsoft SQL Server, Error: 10060)|
+|10054|The data value for one or more columns overflowed the type used by the provider.|
+|10053|Could not convert the data value due to reasons other than sign mismatch or overflow.|
+|997|A connection was successfully established with the server, but then an error occurred during the login process. (provider: Named Pipes Provider, error: 0 - Overlapped I/O operation is in progress)|
+|233|A connection was successfully established with the server, but then an error occurred during the login process. (provider: Shared Memory Provider, error: 0 - No process is on the other end of the pipe.) (Microsoft SQL Server, Error: 233)|
+|64||
```

Comment:
No descrption for `64`, `20`, and `0` ?

Reply:
Thank you for bringing it up. I couldn't find an acceptable description for these error numbers, especially error number 0.
Any suggestions @David-Engel @cheenamalhotra?

Reply:
@DavoudEshtehari Did these error numbers come from Silvano?

Reply:
Except number 997, all them come from Silvano.

Reply:
For error 0, below exceptions are seen:

- `Microsoft.Data.SqlClient.SqlException (0x80131904): A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible.
Verify that the instance name is correct and that SQL Server is configured to allow remote connections.
(provider: TCP Provider, error: 0 - No such host is known.)`

This is retriable as the message "No such host is known` is known retriable error.

- `A connection was successfully established with the server, but then an error occurred during the login process.
(provider: SSL Provider, error: 0 - The target principal name is incorrect.)`

This is an error generated by Domain controller during authentication. More info: https://docs.microsoft.com/en-us/troubleshoot/windows-server/identity/target-principal-name-is-incorrect-when-replicating-data
No info on retry options.

- ![image](https://user-images.githubusercontent.com/13396919/109865169-98d3aa80-7c18-11eb-9a6c-d83362673187.png)
![image](https://user-images.githubusercontent.com/13396919/109865223-aa1cb700-7c18-11eb-958a-b0c35807862a.png)

They are server timeout errors, which are also retriable.

**Summarizing**
Error 0 is very generic number, and should ideally not be retried upon. But looking at errors that arise with it, it may be okay to retry on.